core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/inputEncoder.kt

```diff
@@ -0,0 +1,48 @@
+package org.evomaster.core.problem.rest.classifier
+
+import org.evomaster.core.problem.rest.data.RestCallAction
+import org.evomaster.core.search.gene.BooleanGene
+import org.evomaster.core.search.gene.collection.EnumGene
+import org.evomaster.core.search.gene.numeric.DoubleGene
+import org.evomaster.core.search.gene.numeric.IntegerGene
+
+
+// Encoding IntegerGene, DoubleGene, EnumGene, and BooleanGene of a RestCallAction into a double vector
+fun inputEncoder(input: RestCallAction): List<Double> {
+
+    // Create enum encodings
+    val enumEncoding = mutableMapOf<String, Map<String, Int>>()
+    for (parameter in input.parameters) {
```

Comment:
you might want to iterate of sorted parameters, by name, to avoid non-deterministic behavior

Reply:
also, for now, might just want to consider `QueryParam`